rhombus/private/syntax-meta.rkt

```diff
@@ -48,40 +48,39 @@
   (define/arity (syntax_meta.value id/op
                                    [sp expr-space-path]
                                    [fail (lambda ()
-                                           (raise-syntax-error 'syntax_meta.value "no binding" id/op))])
-    (define id (extract-name 'syntax_meta.value id/op sp))
+                                           (raise-syntax-error who "no binding" id/op))])
+    (define id (extract-name who id/op sp))
     (syntax-local-value id (if (and (procedure? fail)
                                     (procedure-arity-includes? fail 0))
                                fail
                                (lambda () fail))))
 
-  (define (extract-free-name stx sp)
-    (extract-name 'syntax_meta.equal_binding stx sp #:build-dotted? #t))
+  (define/arity (syntax_meta.equal_binding id1
+                                           id2
+                                           [sp expr-space-path]
+                                           [phase1 (syntax-local-phase-level)]
+                                           [phase2 phase1])
+    (define (extract-free-name stx sp)
+      (extract-name who stx sp #:build-dotted? #t))
+    (free-identifier=? (extract-free-name id1 sp)
+                       (extract-free-name id2 sp)
+                       phase1
+                       phase2))
 
-  (define/arity syntax_meta.equal_binding
-    (case-lambda
-      [(id1 id2) (free-identifier=? (extract-free-name id1 expr-space-path) (extract-free-name id2 expr-space-path))]
-      [(id1 id2 sp) (free-identifier=? (extract-free-name id1 sp) (extract-free-name id2 sp))]
-      [(id1 id2 sp phase1) (free-identifier=? (extract-free-name id1 sp) (extract-free-name id2 sp) phase1)]
-      [(id1 id2 sp phase1 phase2) (free-identifier=? (extract-free-name id1 sp) (extract-free-name id2 sp) phase1 phase2)]))
```

Comment:
This function was written with `case-lambda` to avoid calling `(syntax-local-phase-level)`, relying instead on the default in `free-identifier=?` as a kind of shortcut.